components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -809,20 +803,25 @@ void AIChatService::OnClientConnectionChanged(ConversationHandler* handler) {
   MaybeUnloadConversation(handler);
 }
 
-void AIChatService::OnConversationTitleChanged(ConversationHandler* handler,
-                                               std::string title) {
-  auto conversation_it = conversations_.find(handler->get_conversation_uuid());
-  CHECK(conversation_it != conversations_.end());
-  auto& conversation = conversation_it->second;
-  conversation->title = title;
+void AIChatService::OnConversationTitleChanged(
+    const std::string& conversation_uuid,
+    const std::string& new_title) {
+  auto conversation_it = conversations_.find(conversation_uuid);
+  if (conversation_it == conversations_.end()) {
+    DLOG(ERROR) << "Conversation not found for title change";
+    return;
+  }
+
+  auto& conversation_metadata = conversation_it->second;
+  conversation_metadata->title = new_title;
 
   OnConversationListChanged();
 
   // Persist the change
   if (ai_chat_db_) {
     ai_chat_db_
         .AsyncCall(base::IgnoreResult(&AIChatDatabase::UpdateConversationTitle))
-        .WithArgs(handler->get_conversation_uuid(), std::move(title));
+        .WithArgs(conversation_uuid, new_title);
```

Comment:
if we just take a `std::string`:

```suggestion
  auto& conversation_metadata = conversation_it->second;
  conversation_metadata->title = std::move(new_title);

  OnConversationListChanged();

  // Persist the change
  if (ai_chat_db_) {
    ai_chat_db_
        .AsyncCall(base::IgnoreResult(&AIChatDatabase::UpdateConversationTitle))
        .WithArgs(conversation_uuid, conversation_metadata->title);
```